### DISCUSSION TITLE
Github & Atom: Unable to push on Atom (Couldn't connect to server)

### DISCUSSION THREAD

**Discussion by mushroomspores** [community/community#147049] [New to GitHub]
I just encountered this error message when I tried to push some staged changes on the Atom desktop app:

> fatal: unable to access 'https://github.com/mushroomspores/angelicaong/': Failed to connect to 127.0.0.1 port 49189 after 0 ms: Couldn't connect to server

For context, I'm using Github alongside Atom. I figured that I encountered this error possibly because I'm not logged in to Github on the Atom desktop app, so I visited github.atom.io/login to generate an authentication token but was instead redirected to a notification about Atom having been sunset in 2022. I did not manage to update my desktop app to the latest before then (I'm still on 1.58.1).

Is there a way for me to still get that authentication token so I can log in and make updates to my site? I love the Atom interface and am really a complete noob at coding (I just know enough basic HTML to get my site running) so I'm a little afraid of switching to a completely new and unfamiliar app, as I would then have to somehow Google my way through to figure out how to set everything all up so it all works.

I parked this under "New to Github" even though I've been using it for a few years because I really don't know anything about how it works, and just used the few features that I was taught to use (it was a literal step-by-step tutorial that I just faithfully followed). I don't have a background in software engineering or coding so a lot of the words I see appear as technical jargon to me, which are a little intimidating. 

I'd appreciate all the guidance I can get. Thank you.

**Comment by nishanmudalige** [comment]
Have you tried coding with atom, however pushing updates to the site using a separate terminal window?

**Comment by chtholine** [comment]
Since Atom has been sunset and is no longer receiving updates or support, you won’t be able to generate a new authentication token through the Atom interface. However, you can still push your changes to GitHub using other methods. Here’s a step-by-step guide on how to handle this situation:

### Steps to Push Changes Without Atom:

1. **Generate a Personal Access Token (PAT) on GitHub**:
   - Go to [GitHub Personal Access Tokens](https://github.com/settings/tokens).
   - Click "Generate new token."
   - Select the necessary scopes (e.g., `repo` for full control of private repositories).
   - Generate the token and copy it (you won’t be able to view it again).

2. **Use Git Command Line**:
   - Open your terminal or command prompt.
   - Navigate to your project directory using `cd` (e.g., `cd path/to/your/project`).
   - Run the following commands to configure your repository to use your PAT:
     ```bash
     git remote set-url origin https://<USERNAME>@github.com/mushroomspores/angelicaong.git
     ```
     Replace `<USERNAME>` with your GitHub username.
   - Push your changes:
     ```bash
     git push
     ```
   - When prompted for your password, paste the PAT instead.

3. **Consider Using a New Code Editor**:
   - Since Atom is no longer supported, consider switching to an actively maintained editor like **Visual Studio Code (VS Code)**. It has a similar interface and excellent GitHub integration.
   - Install VS Code from [here](https://code.visualstudio.com/).
   - Open your project in VS Code and use the built-in source control feature to push changes to GitHub.

### Additional Tips:
- **Learn Git Basics**: Familiarizing yourself with basic Git commands can help you manage repositories without relying solely on a GUI.
- **Use GitHub Desktop**: If you prefer a GUI, GitHub Desktop is a beginner-friendly alternative for managing repositories. Download it from [GitHub Desktop](https://desktop.github.com/).
- **Ask for Help**: Don’t hesitate to seek help on forums like Stack Overflow or GitHub Discussions for any specific issues you face.

Switching tools might seem intimidating, but VS Code and GitHub Desktop are user-friendly and have vast community support to help you through the transition.

**Reply by ghostinhershell** [reply]
Hi @mushroomspores 👋🏾,It looks like @chtholine's response was the answer to your question here. Can you please mark their response as the answer? When you mark a question as an answer, GitHub will highlight the comment and replies to the comment to help visitors quickly find the answer.
![image](https://github.com/user-attachments/assets/b89e7161-c2d1-47e2-a058-242df633e6b2).
